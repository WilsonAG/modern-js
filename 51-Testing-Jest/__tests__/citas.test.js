import Citas from '../js/classes/Citas';

describe('Probar clase de citas', () => {
  const citas = new Citas();
  const id = Date.now()

  test('Agregar una nueva cita', () => {
    const citaObj = {
      id,
      mascota: 'Connie',
      propietario: 'Andrea',
      telefono: '2692458',
      fecha: '06-06-2021',
      hora: '10:30',
      sintomas: 'Solo duerme',
    };

    citas.agregarCita(citaObj)

    expect(citas).toMatchSnapshot()
  });

  test('Actualizar cita', () => {
    const citaActualizada = {
      id,
      mascota: 'Connie Actualizado',
      propietario: 'Andrea',
      telefono: '2692458',
      fecha: '06-06-2021',
      hora: '10:30',
      sintomas: 'Solo duerme',
    };

    citas.editarCita(citaActualizada)

    expect(citas).toMatchSnapshot()
  })

  test('Eliminar cita', () => {
    citas.eliminarCita(id)

    expect(citas).toMatchSnapshot()
  })

});
