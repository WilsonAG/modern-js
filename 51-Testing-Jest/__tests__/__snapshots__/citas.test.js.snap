// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "06-06-2021",
      "hora": "10:30",
      "id": 1623001460011,
      "mascota": "Connie Actualizado",
      "propietario": "Andrea",
      "sintomas": "Solo duerme",
      "telefono": "2692458",
    },
  ],
}
`;

exports[`Probar clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "06-06-2021",
      "hora": "10:30",
      "id": 1623001460011,
      "mascota": "Connie",
      "propietario": "Andrea",
      "sintomas": "Solo duerme",
      "telefono": "2692458",
    },
  ],
}
`;

exports[`Probar clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
